### ISSUE TITLE
Bluetooth Low Energy Question

### ISSUE THREAD

**Issue by sshome97** [chickenjohn/ECG_Monitor_on_Android#5]
Hello, 
Is this code compatible with BLE? Right now we're unable to connect the application to our microcontroller(NRF51822)

**Comment by chickenjohn** [comment]
I don't think it will support BLE device. You could implement your own BluetoothManager class to support it. PR is welcomed!

**Comment by sshome97** [comment]
Thanks for the quick response. Another question I had is  what does the SendWaveTool and how would I implement it? I'm assuming its used as sample data to show the fully operational application?

Thanks

**Comment by chickenjohn** [comment]
SendWaveTool is a python script I used on my computer to mimic the ECG signal sending from host to the Android App. As the host, my computer sends leadii.bin out via serial port to a [bluetooth serial port device. ](https://www.ebay.com/itm/BT06-HC-06-Bluetooth-Serial-Port-Wireless-Data-Module-Support-UART-Port-New-US/272747239492?hash=item3f81009844:g:wmAAAOSwjvJZWsFD) The leadii.bin is from [MIT-BIH Database](https://physionet.org/content/mitdb/1.0.0/), but in binary form.

**Comment by sshome97** [comment]
Thanks again for the quick response. We are trying to test your application, but we don't understand how to display the ECG waveform? Meaning how will the Bluetooth serial port device lead the leadii.bin? Is this code separate from the code in your project (ie microcontroller code)?
Also, where is the call in your code for the application to read data being transmitted from the bluetooth device?

Thanks

**Comment by chickenjohn** [comment]
For your first question: In my experiment, I connected the bluetooth serial port to my laptop via USB-to-ttl converter(like [this](https://www.ebay.com/itm/USB-To-RS232-TTL-PL2303HX-Cable-Adapter-COM-Module-Auto-Converter-for-Arduino/302749462289?hash=item467d462f11:g:KwUAAOSw8KRcXakd)). With this I am able to send the leadii.bin through USB->serial port->bluetooth device by running the sendwavetool.py

For your second question: [this thread](https://github.com/chickenjohn/ECG_Monitor_on_Android/blob/6abbac8aaf7f43f1241031a9a86ad6a460ad7155/app/src/main/java/com/experiment/chickenjohn/materialdemo/bluetoothManager.java#L121) is always listening to the bluetooth rfcomm socket

**Comment by sshome97** [comment]
Hi again, 
Your RR interval analysis seems very interesting. But I don't really understand how it works. Would you mind explaining it to me (Line 78 your Run thread). Our current code implements BLE where we receive data and store it in our array. We then want to send this array to a function similar to your ECGDataAnalyzer to determine the RR interval. I noticed that the RR interval depends on the recordRate. What is this value?
